### PR TITLE
[Chef-16] Fix solaris build and test failures in adhoc pipeline

### PR DESCRIPTION
**Pull request by neha-p6**
Signed-off-by: Neha Pansare <neha.pansare@progress.com>

## Description
This change backports fixes here https://github.com/chef/chef/pull/13230/, adding as new commits instead of cherrypick as there are multiple commits

Solaris builds are failing with error
```
export/home/chef-ci/builds/chef/chef-chef-chef-16-omnibus-adhoc/omnibus/vendor/bundle/ruby/3.1.0/bundler/gems/omnibus-83d8428abde4/lib/omnibus/health_check.rb:108:in `block (2 levels) in run!': undefined method `include?' for nil:NilClass (NoMethodError)
 
          bad_omnibus_libs, bad_omnibus_bins = bad_libs.keys.partition { |k| k.include? "embedded/lib" }
                                                                              ^^^^^^^^^`
```
 This change updates the omnibus SHA in omnibus/Gemfile.lock so as to pull in those fixes from 
 (https://github.com/chef/omnibus/pull/1083)
 (https://github.com/chef/omnibus/pull/1082)

Details about test failures are [here](https://chefio.atlassian.net/browse/INFC-274)                                                                            

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by neha-p6** [comment]
Adhoc build testing this change https://buildkite.com/chef/chef-chef-chef-16-omnibus-adhoc/builds/51

File: spec/functional/shell_spec.rb [hunk 1]
```diff
@@ -80,6 +80,12 @@ def run_chef_shell_with(options)
       # so hide the require here
 
       require "pty"
+
+      # FIXME this is temporary... Solaris envs have TERM set to unknown
+      # and the value isn't propagating from the build environment TERM
+      # variable
+      ENV["TERM"] = "vt100" if ["", "unknown"].include?(ENV["TERM"].to_s)
```
Comment (jaymzh) [inline spec/functional/shell_spec.rb:87]: Lets fix the trailing whitespace here?

**Review by jaymzh — CHANGES_REQUESTED** [review]
Minor change to not further break builds/tests.

File: spec/functional/shell_spec.rb (hunk 1 above)
Comment (neha-p6) [reply spec/functional/shell_spec.rb:87]: Done.

**Review by johnmccrae — APPROVED** [review]

**Pull request merged**